src/counts.jl

```diff
@@ -283,6 +283,40 @@ function addcounts_dict!(cm::Dict{T}, x::AbstractArray{T}) where T
     return cm
 end
 
+"""Specialist addcounts methods for small bits types"""
+function addcounts!(cm::Dict{Bool}, x::AbstractArray{Bool})
+    sumx = sum(x)
+    cm[true] = sumx
+    cm[false] = length(x) - sumx
+    cm
+end
+
+"""toindex functions to convert an integer to an array index between 1 to typemax{T}"""
+function toindex(x::Int16)
+  Int(x) + 32769
+end
+
+function toindex(x::Int8)
+  Int(x) + 129
+end
+
+function toindex(x::T) where T <: Union{UInt8, UInt16}
+  Int(x) + 1
+end
+
+function addcounts!(cm::Dict{T}, x::Vector{T}) where T <: Union{UInt8, UInt16, Int8, Int16}
+  arr = zeros(Int, 2^(sizeof(T)*8))
```

Comment:
Maybe better call this `counts`? Below, `arr1` would better be e.g. `c`, as it's not an array.